ui/selectors/selectors.js

```diff
@@ -1573,6 +1589,43 @@ export const getMemoizedUnapprovedTypedMessages = createDeepEqualSelector(
   (unapprovedTypedMessages) => unapprovedTypedMessages,
 );
 
+/**
+ * Get the display name for an address.
+ * This selector will look into the internal accounts and address book to find a display name for the address.
+ *
+ * @param {object} state - The Redux state object.
+ * @param {string} address - The address to get the display name for.
+ * @returns {string} The display name for the address.
+ */
+export const getAddressDisplayName = (state, address) => {
+  const {
+    address: caipAddress,
+    chain: { namespace, reference },
+  } = parseCaipAccountId(
+    isHexString(address) ? `eip155:1:${address}` : address,
```

Comment:
Ideally this selector just supports CAIP-10 addresses and we can handle this edge-case elsewhere, e.g. in `SnapUIAddress`. Most components should require CAIP-10 and not have backwards compat